lib/conncache.c

```diff
@@ -141,7 +142,11 @@ static char *hashkey(struct connectdata *conn)
   else
     hostname = conn->host.name;
 
-  return aprintf("%s:%ld", hostname, conn->port);
+  DEBUGASSERT(len > 32);
+
+  /* put the number first so that the hostname gets cut off if too long */
+  snprintf(buf, len, "%ld%s", conn->port, hostname);
```

Comment:
That's one of the reasons why have our own set of *printf() implementations and snprintf comes from here: https://github.com/curl/curl/blob/master/lib/curlx.h#L105

If you check the code closer you'll see that we already use snprintf() in numerous places thanks to this.